### PR TITLE
[ELB] Implement `monitors` v3 pkg

### PR DESCRIPTION
**Pull request by lego963**
### What this PR does / why we need it
Implement Dedicated Load Balancer `monitors` v3 pkg

### Which issue this PR fixes
Refers to: #248

### Special notes for your reviewer
```
=== RUN   TestMonitorList
--- PASS: TestMonitorList (1.98s)
=== RUN   TestMonitorLifecycle
    helpers.go:18: Attempting to create ELBv3 LoadBalancer
    helpers.go:56: Created ELBv3 LoadBalancer: b3b3002e-506d-4360-88bc-8ff4d1479b9f
    helpers.go:149: Attempting to create ELBv3 Pool
    helpers.go:164: Created ELBv3 Pool: 6f1b842f-4290-4bf7-950c-7aa443e5331e
    monitors_test.go:38: Attempting to Create ELBv3 Monitor
    monitors_test.go:61: Created ELBv3 Monitor: 3b7b2aa5-5a98-4a14-b6e1-c8d857fc4280
    monitors_test.go:63: Attempting to Update ELBv3 Monitor
    monitors_test.go:72: Updated ELBv3 Monitor: 3b7b2aa5-5a98-4a14-b6e1-c8d857fc4280
    monitors_test.go:52: Attempting to Delete ELBv3 Monitor: 3b7b2aa5-5a98-4a14-b6e1-c8d857fc4280
    monitors_test.go:55: Deleted ELBv3 Monitor: 3b7b2aa5-5a98-4a14-b6e1-c8d857fc4280
    helpers.go:170: Attempting to delete ELBv3 Pool: 6f1b842f-4290-4bf7-950c-7aa443e5331e
    helpers.go:173: Deleted ELBv3 Pool: 6f1b842f-4290-4bf7-950c-7aa443e5331e
    helpers.go:62: Attempting to delete ELBv3 LoadBalancer: b3b3002e-506d-4360-88bc-8ff4d1479b9f
    helpers.go:65: Deleted ELBv3 LoadBalancer: b3b3002e-506d-4360-88bc-8ff4d1479b9f
--- PASS: TestMonitorLifecycle (10.17s)
PASS

Process finished with the exit code 0
```

### CODE REVIEW

File: openstack/elb/v3/monitors/requests.go [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+	})
+}
+
+// Constants that represent approved monitoring types.
+const (
+	TypePING  = "PING"
+	TypeTCP   = "TCP"
+	TypeHTTP  = "HTTP"
+	TypeHTTPS = "HTTPS"
+)
```
Comment (outcatcher) [inline openstack/elb/v3/monitors/requests.go]: Please use `MonitorType` for that constants and later in `CreateOpts` and `Monitor` structs

File: openstack/elb/v3/monitors/requests.go [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+	url := rootURL(client)
+	if opts != nil {
+		query, err := opts.ToMonitorListQuery()
+		if err != nil {
+			return pagination.Pager{Err: err}
+		}
+		url += query
+	}
+	return pagination.NewPager(client, url, func(r pagination.PageResult) pagination.Page {
+		return MonitorPage{pagination.LinkedPageBase{PageResult: r}}
```
Comment (outcatcher) [inline openstack/elb/v3/monitors/requests.go]: This doesn't seems to be a linked page

File: openstack/elb/v3/monitors/requests.go [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+
+	// Maximum number of seconds for a Monitor to wait for a ping reply
+	// before it times out. The value must be less than the delay value.
+	Timeout int `json:"timeout,omitempty"`
+
+	// Number of permissible ping failures before changing the member's
+	// status to INACTIVE. Must be a number between 1 and 10.
+	MaxRetries int `json:"max_retries,omitempty"`
+
+	MaxRetriesDown int `json:"max_retries_down,omitempty"`
```
Comment (outcatcher) [inline openstack/elb/v3/monitors/requests.go:194]: Maybe join fields, if there is no comment for `MaxRetriesDown`?

File: openstack/elb/v3/monitors/results.go [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+	var s struct {
+		Links []golangsdk.Link `json:"healthmonitors_links"`
+	}
+
+	err := r.ExtractInto(&s)
+	if err != nil {
+		return "", err
+	}
+
+	return golangsdk.ExtractNextURL(s.Links)
```
Comment (outcatcher) [inline openstack/elb/v3/monitors/results.go]: Are there such links in the doc?

**Review by outcatcher — CHANGES_REQUESTED** [review]

**Review by outcatcher — DISMISSED** [review]

**Review by outcatcher — APPROVED** [review]

**Review by anton-sidelnikov — APPROVED** [review]

**Pull request merged**